src/common/HTTPRequest.ts

```diff
@@ -217,7 +218,7 @@ export class HTTPRequest {
   /**
    * @returns the most recent reason for aborting the request
    */
-  abortErrorReason(): Protocol.Network.ErrorReason {
+  abortErrorReason(): Protocol.Network.ErrorReason | undefined {
```

Comment:
I am not sure it warrants a major semver update because it's a bug in types. The underlying value always had a chance to be undefined.

Reply:
This is also true and thinking that this is a pretty small change, it might not warrant a major semver update. However, there is a possibility that this change will break people's builds. Soooo I don't know how to move forward, wdyt?

Reply:
I'd be for just making a change.  But we can make the runtime error instead if the underyling value is undefined. I'd prefer a build time error though as it is easier for the users to address.

Reply:
If we think this is not a frequent use case, I'd prefer runtime error since it will affect less users (none if this case is really not possible)

Otherwise, regardless of the people using this API correctly or not, they have a chance to get a build error which possibly will affect more people. WDYT?

Reply:
We don't try to minimize the number of users affected by this. The ultimate goal is the correctness so that the users can rely on our API. I think here we actually need to change the API as the abort reason is not guaranteed to be defined previously.

Reply:
I will mark the change accordingly.

Reply:
Hihiiim makes sense to change the API in a better way though I'm still having second thoughts of breaking people's builds for this reason 😅

Say I, as a user, had this code:
```
function printErrorReason(errorReason: ErrorReason) {
    console.log(errorReason);
}
const errorReason = httpRequest.abortErrorReason();
printErrorReason(errorReason);
```

Now, when I upgrade Puppeteer a minor version that contains the changes in this PR, it will break my build for this code since `printErrorReason` expects a call with an `ErrorReason` and not `ErrorReason | null`. I don't have strong opinions on whether this breakage is acceptable or not, so if you think it is acceptable, we can move forward with the API change.